cpp/ql/lib/semmle/code/cpp/commons/Printf.qll

```diff
@@ -268,6 +270,18 @@ class FormattingFunctionCall extends Expr {
   }
 }
 
+/**
+ * Gets the number of digits required to represent the integer represented by `f`.
+ *
+ * `f` is assumed to be nonnegative.
+ */
+bindingset[f]
+private int lengthInBase10(float f) {
+  f = 0 and result = 0
+  or
+  result = f.log10().ceil()
```

Comment:
```suggestion
  result = f.log10().floor() + 1
```
I believe this will be correct for exact powers of 10.

Reply:
Well spotted! I've included this fix in 9a9f7943aa4e7a1d1f954c470b2f7ea574c85771.